tcpclient/connectionsManagement.go

```diff
@@ -44,15 +48,27 @@ func TCPConnect(id int, host string, port int, wg *sync.WaitGroup, debugOut io.W
 	reportConnectionStatus(debugOut, statusChannel, connectionDescription)
 	connBuf := bufio.NewReader(conn)
 	for {
-		str, err := connBuf.ReadString('\n')
-		if len(str) > 0 {
-			fmt.Fprintln(debugOut, "Connection", id, "got", str)
-		}
-		if err != nil {
-			connectionDescription.status = ConnectionClosed
-			reportConnectionStatus(debugOut, statusChannel, connectionDescription)
-			wg.Done()
-			return err
+		select {
+		case <-closeRequest:
+			// we need to close gracefully the connection
+		default:
+			// TODO: to review
+			conn.SetReadDeadline(time.Now().Add(time.Duration(defaultReadTimeoutInSecs) * time.Second))
+			str, err := connBuf.ReadString('\n')
+			if len(str) > 0 {
+				fmt.Fprintln(debugOut, "Connection", id, "got", str)
+			}
+			if err, ok := err.(net.Error); ok && err.Timeout() {
```

Comment:
should `err` be used before `str`?

Reply:
I would also use another variable name like the used in golang documentation to make it distinctly from `err`:
```
if nerr, ok := err.(net.Error); ok && nerr.Temporary() {
    time.Sleep(1e9)
    continue
}
```

Reply:
in fact, in the newer version that was already there i actually opted by terror :) . I'd change the order, for clarity (it still works as errored responses do not generate a response length bigger than 0)